nuxeo-importer-xml-parser/src/main/java/org/nuxeo/ecm/platform/importer/xml/parser/XMLImporterServiceImpl.java

```diff
@@ -328,8 +329,13 @@ protected void processDocAttributes(DocumentModel doc, Element el, AttributeConf
                     }
                 }
             } else {
-                value = (Serializable) resolveComplex(el, conf);
-                if (value != null && !conf.getMapping().isEmpty()) {
+                Map<String, Object> props = (Map<String, Object>) resolveComplex(el, conf);
+                value = new HashMap<>(props);
+                if (props.containsKey("file")) {
+                    Blob blob = resolveBlob(el, conf, "file");
+                    props.put("file", blob);
```

Comment:
Please use constants for `"file"`. Use several constants if they have different roles (I don't know from just reading the diff).